### PR TITLE
Feature zeebe component

### PR DESCRIPTION
**Pull request by LoRez**
# Description

This is a new component to support Camunda Zeebe (from version 8) within camel.

# Target

- [ X] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [X ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [X ] I checked that each commit in the pull request has a meaningful subject line and body.

- [X ] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java
Comment (orpiske) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java:66]: I think this should be `LOG.error("Invalid Operation {}", operationName.value());`

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java
Comment (orpiske) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java:61]: I think `ObjectHelper.notNull` would do the trick here. That's what we normally use.

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java
Comment (orpiske) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java:101]: A few things:

1. Maybe add a message explaining what is being logged.
2. Also, seems like it would be better to use a lower debug level (`debug`, maybe?).

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java
Comment (orpiske) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java:74]: The `String.format` seems unnecessary here. Also, a bit more details about the state of `operationName` would be helpful (i.e.; "Invalid operation: null operation name"); 

Lastly, probably a better check w/ the `ObjectHelper.notNull`.

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java
Comment (orpiske) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java:83]: Same as above.

File: components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ResourceDeploymentIntegrationTest.java
Comment (orpiske) [inline components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ResourceDeploymentIntegrationTest.java:131]: IIRC, there's an assert specific to check for the type.

File: components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ZeebeConsumerIntegrationTest.java
Comment (orpiske) [inline components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ZeebeConsumerIntegrationTest.java:89]: Please, avoid `sleeps` on the test code. Our tests already consume way too much time and we have been trying to reduce this. If some kind of synchronization is needed, I'd recommend looking at using Awaitility to do so.

File: components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ZeebeConsumerIntegrationTest.java
Comment (orpiske) [inline components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ZeebeConsumerIntegrationTest.java:112]: It would be good to have a fail message here.

File: components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ZeebeConsumerIntegrationTest.java
Comment (orpiske) [inline components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ZeebeConsumerIntegrationTest.java:126]: Same note about `sleeps` as I raised before.

File: components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ZeebeConsumerIntegrationTest.java
Comment (orpiske) [inline components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ZeebeConsumerIntegrationTest.java:151]: Same note about `fail` messages.

File: components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/model/DeploymentRequestTest.java
Comment (orpiske) [inline components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/model/DeploymentRequestTest.java:44]: Please use `Assert.assertDoesNotThrow`  whenever you need to assert some piece of test code does not throw an exception.

File: components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/model/DeploymentRequestTest.java
Comment (orpiske) [inline components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/model/DeploymentRequestTest.java:57]: Same as above.

**Review by orpiske — CHANGES_REQUESTED** [review]
Thanks for your contribution! 

I have left some comments about issues that need to be fixed. 

Also waiting on a few more folks to review the code.

File: components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ResourceDeploymentIntegrationTest.java
Comment (LoRez) [reply components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ResourceDeploymentIntegrationTest.java:131]: assertObjectIsClass does not exist in JUnit5. It seems it was removed. Stackoverflow recommends instances

**Comment by LoRez** [comment]
Did resolve the feedback, thanks a lot. One point cannot be resolved as recommended as assertObjectIsClass does not exist in JUnit5.

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java
Comment (LoRez) [reply components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java]: Done

File: components/camel-zeebe/pom.xml
Comment (orpiske) [inline components/camel-zeebe/pom.xml:103]: Our standard is to name Integration tests as `SomeThingIT.java` (i.e.: **IT.java). Our default surefire configuration is set to skip that and the failsafe is configured to include that. This also impacts a bit the code analysis on Sonar. 

Please, can you rename the integration tests to match this, so we avoid this extra code?

Comment (orpiske) [reply components/camel-zeebe/pom.xml:103]: Obs.: it's component-specific configurations for surefire and failsafe are OK to keep if they are needed.

File: components/camel-zeebe/src/main/docs/zeebe-component.adoc
Comment (orpiske) [inline components/camel-zeebe/src/main/docs/zeebe-component.adoc:34]: This seems a bit out of order. The dependency is listed after the prerequisites.

**Review by orpiske — CHANGES_REQUESTED** [review]
Thanks for the changes! I think you are on the right track. I've left a few more comments.

One last tip is regarding one change that could make this code easier to port to Camel 4 ... 

In a few places you are using `exchange.getIn()` or `exchange.getOut()`. These are deprecated and with Camel 4 there is going to be only `exchange.getMessage()`. 

Of course, this is not a blocking issue at the moment, but it would make our life easier when/if we cherry-pick the code to `main` 😅.

**Review by oscerd — CHANGES_REQUESTED** [review]
This will need to be merged on main too and we need also to ensure the component will be added to kit

File: components/camel-zeebe/pom.xml
Comment (LoRez) [reply components/camel-zeebe/pom.xml:103]: I tried renaming the integration tests to IT but then I was not able to disable them through the profile and they failed when the service is not present, which I guess will be the case for almost everybody.
Is there a way to name them IT and insure that they are only executed when the zeebe-integration profile is active?

File: components/camel-zeebe/src/main/docs/zeebe-component.adoc
Comment (LoRez) [reply components/camel-zeebe/src/main/docs/zeebe-component.adoc:34]: It seems there are multiple flavors how adocs are structured. I changed it so that the pom info is at the end as in the redis component.

File: components/camel-zeebe/pom.xml
Comment (orpiske) [reply components/camel-zeebe/pom.xml:103]: You can use JUnit's `@DisabledIfSystemProperty` or `@EnabledIfSystemProperty`. 

Here's an example: https://github.com/apache/camel/blob/main/components/camel-kafka/src/test/java/org/apache/camel/component/kafka/integration/health/KafkaConsumerBadPortHealthCheckIT.java#L61-L62

File: components/camel-zeebe/src/main/docs/zeebe-component.adoc
Comment (orpiske) [reply components/camel-zeebe/src/main/docs/zeebe-component.adoc]: Thanks. That works.

File: components/camel-zeebe/pom.xml
Comment (LoRez) [reply components/camel-zeebe/pom.xml]: Ok, changed the behavior to use EnbaledIfSystemProperty. Works fine. Thank you

**Review by orpiske — APPROVED** [review]
Thanks for you contribution, @LoRez. I think you solved the comments I raised. Giving my +1 here. Let's wait for a few more reviewers to take a look.

**Comment by LoRez** [comment]
> This will need to be merged on main too and we need also to ensure the component will be added to kit

Just to be sure. Should I create a second PR where the code is merged into main?

**Comment by orpiske** [comment]
> > This will need to be merged on main too and we need also to ensure the component will be added to kit
> 
> Just to be sure. Should I create a second PR where the code is merged into main?

Yes. Once this one is merged, then I think you can create another one for the `main` branch, so it goes to Camel 4 as well (if possible by the component, supports java 17, supports Jakarta, etc etc etc).

**Comment by orpiske** [comment]
It's failing the checkstyle check, but I think it's unrelated to this change. 


@LoRez when you get a chance, please, can you rebase the code and push again? 

[checkstyle.log](https://github.com/apache/camel/files/10521104/checkstyle.log)

File: components/camel-zeebe/pom.xml
Comment (davsclaus) [inline components/camel-zeebe/pom.xml:34]: 3.21

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeComponent.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeComponent.java:37]: label = security, secret = true

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeComponent.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeComponent.java:39]: label = security

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeComponent.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeComponent.java:35]: label = security

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java:64]: Do not log and throw exception, only throw exception

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java:74]: Maybe close should be in try .. catch and WARN log if closing failed

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java:98]: Use ifDebugEnabled as toJson is expensive

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java:36]: 3.21.0

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java:37]: Category is JBPM or some sort kind like that

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java:51]: javaType = should only be set when its a special type, so remove

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java:55]: javaType = should only be set when its a special type, so remove

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeProducer.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeProducer.java:57]: Do not log and throw exception, only trow exception

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeProducer.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeProducer.java:49]: Add doInit or doStart and validate that processor != null and fail if so, eg such validation should be in init or start

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/processor/DeploymentProcessor.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/processor/DeploymentProcessor.java:62]: Log and throw!

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/processor/DeploymentProcessor.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/processor/DeploymentProcessor.java:61]: The body can be inputstream and still possible to read into a String so without using camel type converter then this processor is very limited

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/processor/DeploymentProcessor.java
Comment (davsclaus) [inline components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/processor/DeploymentProcessor.java:73]: log and throw!

**Review by davsclaus — CHANGES_REQUESTED** [review]

**Comment by davsclaus** [comment]
I did about 50% review and there is a fair bit of things to correct, so waiting to continue until resolved

File: components/camel-zeebe/pom.xml
Comment (davsclaus) [inline components/camel-zeebe/pom.xml:36]: Upgrade to 8.1.6 (latest GA release)

File: components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java
Comment (LoRez) [reply components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java:74]: It seems that closing cannot fail. When I look at the implementation there is no exception to be thrown.

**Comment by LoRez** [comment]
> I did about 50% review and there is a fair bit of things to correct, so waiting to continue until resolved
> 
> My comments are just minor, but important to keep the component aligned how its done in other components so its similar and easy for us to maintain. We see sometimes dump and run from component donators.

Thank you that is very helpful. I was looking for pointers into how to write a component and looked a lot at other components, but it is difficult to pull everything together when you do it for the first time.

**Comment by LoRez** [comment]
Also did just check a rebase on main and besides changes in the pom for parent version and log4j version everything works fine.

**Comment by LoRez** [comment]
> It's failing the checkstyle check, but I think it's unrelated to this change.
> 
> @LoRez when you get a chance, please, can you rebase the code and push again?
> 
> [checkstyle.log](https://github.com/apache/camel/files/10521104/checkstyle.log)

Yes, the reported files are in jbang, not related to my code...

**Review by davsclaus — APPROVED** [review]

**Comment by LoRez** [comment]
Once this PR is through, I will provide one for main. @oscerd is that ok for you?

**Review by oscerd — APPROVED** [review]

**Comment by oscerd** [comment]
Sure, thanks a lot

**Comment by davsclaus** [comment]
I am doing a big of polish on camel-3.x branch. Can you grab all of that and send a PR against main

**Pull request merged**
